ui/StatusQ/src/StatusQ/Components/private/chart/Library/Library.js

```diff
@@ -0,0 +1,159 @@
+        // Default options
+        events: [
+            "mousemove",
+            "mouseout",
+            "click"
+        ]
+    })
+
+    // QML rendering plugin
+    Chart.plugins.register({
+        afterDraw: function(chart) {
+            chart.canvas.requestPaint()
+        }
+    })
+
+    var EVENTS = {
+       /*chartJS event: QML event*/
+        click: "clicked",
+        mousemove: "positionChanged",
+        mouseenter: "entered",
+        mouseout: "exited",
+        mousedown: "pressed",
+        mouseup: "released",
+        resize: "resized"
+    }
+
+    function createEvent(type, chart, x, y, native, target) {
+        return {
+            type: type,
+            chart: chart,
+            native: native || null,
+            x: x !== undefined ? x : null,
+            y: y !== undefined ? y : null,
+            target: target || null,
+        }
+    }
+
+    // QML platform implementation
+    Chart.helpers.merge(Chart.platform, {
+        addEventListener: function(chart, type, listener) {
+            const mapped = EVENTS[type]
+            if (!mapped) {
+                console.warn("Unsupported event:", type)
+                return
+            }
+
+            const canvas = chart.canvas
+            const qmlHandler = (event) => {
+                listener(createEvent(
+                    type,
+                    chart,
+                    event && event.x,
+                    event && event.y,
+                    event,
+                    canvas))
+            }
+            
+            canvas._eventSource[mapped].connect(qmlHandler)
+            canvas._eventSource.connectedHandlers[listener] = qmlHandler
+        },
+
+        removeEventListener: function(chart, type, listener) {
+            const canvas = chart.canvas
+            if (!canvas._eventSource.connectedHandlers[listener]) {
+                return
+            }
+
+            const mapped = EVENTS[type]
+            const qmlHandler = canvas._eventSource.connectedHandlers[listener]
+            canvas._eventSource[mapped].disconnect(qmlHandler)
+            delete canvas._eventSource.connectedHandlers[listener]
+        }
+    })
+
+    Chart.helpers.merge(Chart.helpers, {
+        color: function(c) {
+            // Implement pattern & gradient if needed
+            return color(c)
+        },
+        getHoverColor: function(c) {
+            let qmlColor = "#000000"
+
+            if (typeof c === 'object') {
+                qmlColor = QQ.Qt.rgba(c.r, c.g, c.b, c.a)
+            } else if (typeof c === 'string') {
+                if (c.startsWith("#")) {
+                    qmlColor = c
+                } else {
+                    qmlColor = color(c).hexString()
+                }
+            }
+
+            const hoverColor = SQTheme.Theme.palette.hoverColor(qmlColor)
+            var jsColor = {
+                r: hoverColor.red,
+                g: hoverColor.green,
+                b: hoverColor.blue,
```

Comment:
Similar here, you need to use `r`, `g` and `b` respectively as per the `color` property (https://doc.qt.io/qt-5/qml-color.html), and multiply by `255`

Reply:
Just realised we need no translation between qml and js. The colours are set in qml and will be consumed in qml Canvas. If no hover color or background color is set, the hoverColor will be the same as the default background color (but it's not a real use-case). Added the `ThemePalette.hoverColor` as a convenience in case the background color is set, but no hover color.